<?xml version="1.0"?>
<robot name="robot_arm">
    <link name="world_frame"/>
    <joint name="world_to_base" type="fixed">
        <parent link="world_frame"/>
        <child link="base_link"/>
        <origin xyz="0.0 0.0 0.5" rpy="0.0 0.0 0.0"/>
    </joint>
    <link name="base_link"/>
    <joint name="r_shoulder_y_joint" type="revolute">
        <parent link="base_link"/>
        <child link="r_shoulder_y_link"/>
        <axis xyz="0 1 0"/>
        <limit lower="-3" upper="3" velocity="50.0" effort="1000.0"/>
        <origin xyz="0.119 -0.195 -0.02" rpy="3.14159 0 0"/>
    </joint>
    <link name="r_shoulder_y_link"/>
    <joint name="r_shoulder_x_joint" type="revolute">
        <parent link="r_shoulder_y_link"/>
        <child link="r_shoulder_x_link"/>
        <axis xyz="1 0 0"/>
        <limit lower="-3" upper="3" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
    <link name="r_shoulder_x_link"/>
    <joint name="r_shoulder_z_joint" type="revolute">
        <parent link="r_shoulder_x_link"/>
        <child link="r_shoulder_z_link"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3" upper="3" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
    <link name="r_shoulder_z_link"/>
    <joint name="r_elbow_y_joint" type="revolute">
        <parent link="r_shoulder_z_link"/>
        <child link="r_elbow_y_link"/>
        <axis xyz="0 1 0"/>
        <limit lower="0" upper="3" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0.26" rpy="0 0 0"/>
    </joint>
    <link name="r_elbow_y_link"/>
    <joint name="r_elbow_z_joint" type="revolute">
        <parent link="r_elbow_y_link"/>
        <child link="r_elbow_z_link"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
    <link name="r_elbow_z_link"/>
    <joint name="r_wrist_x_joint" type="revolute">
        <parent link="r_elbow_z_link"/>
        <child link="r_wrist_x_link"/>
        <axis xyz="1 0 0"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0.26" rpy="0 0 0"/>
    </joint>
    <link name="r_wrist_x_link"/>
    <joint name="r_hand_fixed" type="fixed">
        <parent link="r_wrist_x_link"/>
        <child link="r_hand_link"/>
        <origin xyz="0.0 0.0 0.07" rpy="0.0 0.0 0.0"/>
    </joint>
    <link name="r_hand_link"/>


    <joint name="l_shoulder_y_joint" type="revolute">
        <parent link="base_link"/>
        <child link="l_shoulder_y_link"/>
        <axis xyz="0 1 0"/>
        <limit lower="-3" upper="3" velocity="50.0" effort="1000.0"/>
        <origin xyz="0.119 0.195 -0.02" rpy="3.14159 0 3.14159"/>
    </joint>
    <link name="l_shoulder_y_link"/>
    <joint name="l_shoulder_x_joint" type="revolute">
        <parent link="l_shoulder_y_link"/>
        <child link="l_shoulder_x_link"/>
        <axis xyz="1 0 0"/>
        <limit lower="-3" upper="3" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
    <link name="l_shoulder_x_link"/>
    <joint name="l_shoulder_z_joint" type="revolute">
        <parent link="l_shoulder_x_link"/>
        <child link="l_shoulder_z_link"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3" upper="3" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
    <link name="l_shoulder_z_link"/>
    <joint name="l_elbow_y_joint" type="revolute">
        <parent link="l_shoulder_z_link"/>
        <child link="l_elbow_y_link"/>
        <axis xyz="0 1 0"/>
        <limit lower="0" upper="3" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0.26" rpy="0 0 0"/>
    </joint>
    <link name="l_elbow_y_link"/>
    <joint name="l_elbow_z_joint" type="revolute">
        <parent link="l_elbow_y_link"/>
        <child link="l_elbow_z_link"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
    <link name="l_elbow_z_link"/>
    <joint name="l_wrist_x_joint" type="revolute">
        <parent link="l_elbow_z_link"/>
        <child link="l_wrist_x_link"/>
        <axis xyz="1 0 0"/>
        <limit lower="-3.14159" upper="3.14159" velocity="50.0" effort="1000.0"/>
        <origin xyz="0 0 0.26" rpy="0 0 0"/>
    </joint>
    <link name="l_wrist_x_link"/>
    <joint name="l_hand_fixed" type="fixed">
        <parent link="l_wrist_x_link"/>
        <child link="l_hand_link"/>
        <origin xyz="0.0 0.0 0.07" rpy="0.0 0.0 0.0"/>
    </joint>
    <link name="l_hand_link"/>

</robot>